dotnet-monitor.yml

```diff
@@ -175,6 +177,36 @@ stages:
         - source: 'bin/OSX.arm64.Release'
         testGroup: None
   - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    # Generate a TPN
+    - template: /eng/common/templates/job/job.yml
```

Comment:
Rather than using a separate job just for this, can you include it with the Windows x64 Release job? Maybe add a template parameter called `generateThirdPartyNotice` with a default value of false and set it to true for this one job.